pkgdb/src/buildenv/realise.cc

```diff
@@ -286,6 +294,31 @@ getRealisedPackages( nix::EvalState &                   state,
   /* Interpret ooutput as derivation. */
   auto package_drv = getDerivation( state, *output.value, false );
 
+  auto broken = package_drv->queryMetaBool( "broken", false );
+  if ( broken && ! allows.broken.value_or( false ) )
```

Comment:
question blocking: do we also need to check if the package is marked as optional?

Reply:
The main question i whether we want to check the same guards as locking does at all during build.
Current guards are there, because nix was checking them at build time (as opposed to locking time) but should eventually be moved to locking time.

Reply:
I meant checking optional before throwing. The behavior of this code is to throw an error for:
```
slack.pkg-path = "slack"
slack.optional = true
```

I would think `optional` means we skip any and all errors. We probably need to clarify our docs either way, because they don't make it very clear

Reply:
I though optional meant "try locking, peroit locking failure", if unfree (/broken) are checked during locking they would adhere to optional.
I think I'll look into #492 related changes on the resolution side and move these checks there -- for this PR i rather try focusing on _integrating the fetcher_.
Going into the resolution to address another issue seems like unrelated work.

Reply:
Okay we can address as part of 492